### ISSUE TITLE
NotSupportedException: Specified method is not supported. on setting interpreted array value via class with constructor 

### ISSUE THREAD

**Issue by styrbo** [scottyboy805/dotnow-interpreter#7]
```
public class InterpretetedMono : MonoBehaviour
    {
        private void Start()
        {
            var array1 = new TestClass[1];
            array1[0] = new TestClass("134");
            Debug.Log("Start work!"); //output: Start work!
            
            InitTestArray();
        }

        public void InitTestArray()
        {
            var array2 = new TestClass[1];
            array2[0] = new TestClass("134"); //error here
            Debug.Log("Init work!");
        }

        class TestClass
        {
            public TestClass(string text)
            {
                Debug.Log("construct"); //output: construct
                NameKey = text;
                Debug.Log("construct1"); //output: construct1
            }

            public string NameKey;
        }
    }
```

if i am remove argument **text** from constructor then interpreted code work fine

errors log:

```
At method body: System.Void Game.InterpretetedMono::InitTestArray()
0x00007ff6633c8bdc (Unity) StackWalker::GetCurrentCallstack
0x00007ff6633cffd9 (Unity) StackWalker::ShowCallstack
0x00007ff66412cd9c (Unity) GetStacktrace
0x00007ff664b75e83 (Unity) DebugStringToFile
0x00007ff6634802f6 (Unity) DebugLogHandler_CUSTOM_Internal_Log
0x0000025034e9d30b (Mono JIT Code) (wrapper managed-to-native) UnityEngine.DebugLogHandler:Internal_Log (UnityEngine.LogType,UnityEngine.LogOption,string,UnityEngine.Object)
0x0000025034e9d1ab (Mono JIT Code) UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
0x0000025109dcacee (Mono JIT Code) UnityEngine.Logger:Log (UnityEngine.LogType,object)
0x000002510cf359ea (Mono JIT Code) UnityEngine.Debug:LogError (object)
0x0000025109de4d4b (Mono JIT Code) [ExecutionEngine.cs:182] dotnow.Runtime.ExecutionEngine:Execute (dotnow.AppDomain,dotnow.Runtime.ExecutionFrame,dotnow.Runtime.CIL.CILOperation[],dotnow.Reflection.CLRExceptionHandler[],int,bool) 
0x0000025109de407b (Mono JIT Code) [CLRMethodBodyBase.cs:84] dotnow.Reflection.CLRMethodBodyBase:ExecuteMethodBody (dotnow.Runtime.ExecutionEngine,dotnow.Runtime.ExecutionFrame) 
0x00000250bff2e90b (Mono JIT Code) [CILInterpreter.cs:2646] dotnow.Runtime.CIL.CILInterpreter:ExecuteInterpreted (dotnow.AppDomain,dotnow.Runtime.ExecutionEngine,dotnow.Runtime.ExecutionFrame&,dotnow.Runtime.CIL.CILOperation[]&,dotnow.Reflection.CLRExceptionHandler[]&,dotnow.Runtime.ExecutionEngine/DebugFlags) 
0x0000025109de4b53 (Mono JIT Code) [ExecutionEngine.cs:175] dotnow.Runtime.ExecutionEngine:Execute (dotnow.AppDomain,dotnow.Runtime.ExecutionFrame,dotnow.Runtime.CIL.CILOperation[],dotnow.Reflection.CLRExceptionHandler[],int,bool) 
0x0000025109de407b (Mono JIT Code) [CLRMethodBodyBase.cs:84] dotnow.Reflection.CLRMethodBodyBase:ExecuteMethodBody (dotnow.Runtime.ExecutionEngine,dotnow.Runtime.ExecutionFrame) 
0x000002510cf327cb (Mono JIT Code) [ExecutionMethod.cs:106] dotnow.Runtime.ExecutionMethod:DelegateInvoke (object) 
0x000002510cf32383 (Mono JIT Code) [CLRMethod.cs:295] dotnow.Reflection.CLRMethod/<>c__DisplayClass41_0:<CreateDelegate>b__0 () 
0x000002510cf31c00 (Mono JIT Code) [MonoBehaviourProxy.cs:265] UnityEngine.MonoBehaviourProxy:Start () 
0x00000250bfe39af8 (Mono JIT Code) (wrapper runtime-invoke) object:runtime_invoke_void__this__ (object,intptr,intptr,intptr)
0x00007ffe35c4f1e0 (mono-2.0-bdwgc) [mini-runtime.c:2849] mono_jit_runtime_invoke 
0x00007ffe35bd2ac2 (mono-2.0-bdwgc) [object.c:2921] do_runtime_invoke 
0x00007ffe35bdbb1f (mono-2.0-bdwgc) [object.c:2968] mono_runtime_invoke 
0x00007ff66330cd34 (Unity) scripting_method_invoke
0x00007ff663308011 (Unity) ScriptingInvocation::Invoke
0x00007ff6632dbfad (Unity) MonoBehaviour::InvokeMethodOrCoroutineChecked
0x00007ff6632dc0d8 (Unity) MonoBehaviour::InvokeMethodOrCoroutineChecked
0x00007ff6632d6ce2 (Unity) MonoBehaviour::DelayedStartCall
0x00007ff662e4bfa4 (Unity) DelayedCallManager::Update
0x00007ff663044099 (Unity) `InitPlayerLoopCallbacks'::`2'::EarlyUpdateScriptRunDelayedStartupFrameRegistrator::Forward
0x00007ff66302c4dc (Unity) ExecutePlayerLoop
0x00007ff66302c5b3 (Unity) ExecutePlayerLoop
0x00007ff663033059 (Unity) PlayerLoop
0x00007ff663cefa91 (Unity) PlayerLoopController::UpdateScene
0x00007ff663cdbcef (Unity) PlayerLoopController::EnterPlayMode
0x00007ff663ceb11d (Unity) PlayerLoopController::SetIsPlaying
0x00007ff663cedf87 (Unity) Application::TickTimer
0x00007ff6641334d1 (Unity) MainMessageLoop
0x00007ff664137521 (Unity) WinMain
0x00007ff6657c5752 (Unity) __scrt_common_main_seh
0x00007ffec87f7034 (KERNEL32) BaseThreadInitThunk
0x00007ffec9b82651 (ntdll) RtlUserThreadStart

At instruction: Ldc_I4_1
0x00007ff6633c8bdc (Unity) StackWalker::GetCurrentCallstack
0x00007ff6633cffd9 (Unity) StackWalker::ShowCallstack
0x00007ff66412cd9c (Unity) GetStacktrace
0x00007ff664b75e83 (Unity) DebugStringToFile
0x00007ff6634802f6 (Unity) DebugLogHandler_CUSTOM_Internal_Log
0x0000025034e9d30b (Mono JIT Code) (wrapper managed-to-native) UnityEngine.DebugLogHandler:Internal_Log (UnityEngine.LogType,UnityEngine.LogOption,string,UnityEngine.Object)
0x0000025034e9d1ab (Mono JIT Code) UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
0x0000025109dcacee (Mono JIT Code) UnityEngine.Logger:Log (UnityEngine.LogType,object)
0x000002510cf359ea (Mono JIT Code) UnityEngine.Debug:LogError (object)
0x0000025109de4e73 (Mono JIT Code) [ExecutionEngine.cs:183] dotnow.Runtime.ExecutionEngine:Execute (dotnow.AppDomain,dotnow.Runtime.ExecutionFrame,dotnow.Runtime.CIL.CILOperation[],dotnow.Reflection.CLRExceptionHandler[],int,bool) 
0x0000025109de407b (Mono JIT Code) [CLRMethodBodyBase.cs:84] dotnow.Reflection.CLRMethodBodyBase:ExecuteMethodBody (dotnow.Runtime.ExecutionEngine,dotnow.Runtime.ExecutionFrame) 
0x00000250bff2e90b (Mono JIT Code) [CILInterpreter.cs:2646] dotnow.Runtime.CIL.CILInterpreter:ExecuteInterpreted (dotnow.AppDomain,dotnow.Runtime.ExecutionEngine,dotnow.Runtime.ExecutionFrame&,dotnow.Runtime.CIL.CILOperation[]&,dotnow.Reflection.CLRExceptionHandler[]&,dotnow.Runtime.ExecutionEngine/DebugFlags) 
0x0000025109de4b53 (Mono JIT Code) [ExecutionEngine.cs:175] dotnow.Runtime.ExecutionEngine:Execute (dotnow.AppDomain,dotnow.Runtime.ExecutionFrame,dotnow.Runtime.CIL.CILOperation[],dotnow.Reflection.CLRExceptionHandler[],int,bool) 
0x0000025109de407b (Mono JIT Code) [CLRMethodBodyBase.cs:84] dotnow.Reflection.CLRMethodBodyBase:ExecuteMethodBody (dotnow.Runtime.ExecutionEngine,dotnow.Runtime.ExecutionFrame) 
0x000002510cf327cb (Mono JIT Code) [ExecutionMethod.cs:106] dotnow.Runtime.ExecutionMethod:DelegateInvoke (object) 
0x000002510cf32383 (Mono JIT Code) [CLRMethod.cs:295] dotnow.Reflection.CLRMethod/<>c__DisplayClass41_0:<CreateDelegate>b__0 () 
0x000002510cf31c00 (Mono JIT Code) [MonoBehaviourProxy.cs:265] UnityEngine.MonoBehaviourProxy:Start () 
0x00000250bfe39af8 (Mono JIT Code) (wrapper runtime-invoke) object:runtime_invoke_void__this__ (object,intptr,intptr,intptr)
0x00007ffe35c4f1e0 (mono-2.0-bdwgc) [mini-runtime.c:2849] mono_jit_runtime_invoke 
0x00007ffe35bd2ac2 (mono-2.0-bdwgc) [object.c:2921] do_runtime_invoke 
0x00007ffe35bdbb1f (mono-2.0-bdwgc) [object.c:2968] mono_runtime_invoke 
0x00007ff66330cd34 (Unity) scripting_method_invoke
0x00007ff663308011 (Unity) ScriptingInvocation::Invoke
0x00007ff6632dbfad (Unity) MonoBehaviour::InvokeMethodOrCoroutineChecked
0x00007ff6632dc0d8 (Unity) MonoBehaviour::InvokeMethodOrCoroutineChecked
0x00007ff6632d6ce2 (Unity) MonoBehaviour::DelayedStartCall
0x00007ff662e4bfa4 (Unity) DelayedCallManager::Update
0x00007ff663044099 (Unity) `InitPlayerLoopCallbacks'::`2'::EarlyUpdateScriptRunDelayedStartupFrameRegistrator::Forward
0x00007ff66302c4dc (Unity) ExecutePlayerLoop
0x00007ff66302c5b3 (Unity) ExecutePlayerLoop
0x00007ff663033059 (Unity) PlayerLoop
0x00007ff663cefa91 (Unity) PlayerLoopController::UpdateScene
0x00007ff663cdbcef (Unity) PlayerLoopController::EnterPlayMode
0x00007ff663ceb11d (Unity) PlayerLoopController::SetIsPlaying
0x00007ff663cedf87 (Unity) Application::TickTimer
0x00007ff6641334d1 (Unity) MainMessageLoop
0x00007ff664137521 (Unity) WinMain
0x00007ff6657c5752 (Unity) __scrt_common_main_seh
0x00007ffec87f7034 (KERNEL32) BaseThreadInitThunk
0x00007ffec9b82651 (ntdll) RtlUserThreadStart

At method body: System.Void Game.InterpretetedMono::Start()
0x00007ff6633c8bdc (Unity) StackWalker::GetCurrentCallstack
0x00007ff6633cffd9 (Unity) StackWalker::ShowCallstack
0x00007ff66412cd9c (Unity) GetStacktrace
0x00007ff664b75e83 (Unity) DebugStringToFile
0x00007ff6634802f6 (Unity) DebugLogHandler_CUSTOM_Internal_Log
0x0000025034e9d30b (Mono JIT Code) (wrapper managed-to-native) UnityEngine.DebugLogHandler:Internal_Log (UnityEngine.LogType,UnityEngine.LogOption,string,UnityEngine.Object)
0x0000025034e9d1ab (Mono JIT Code) UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
0x0000025109dcacee (Mono JIT Code) UnityEngine.Logger:Log (UnityEngine.LogType,object)
0x000002510cf359ea (Mono JIT Code) UnityEngine.Debug:LogError (object)
0x0000025109de4d4b (Mono JIT Code) [ExecutionEngine.cs:182] dotnow.Runtime.ExecutionEngine:Execute (dotnow.AppDomain,dotnow.Runtime.ExecutionFrame,dotnow.Runtime.CIL.CILOperation[],dotnow.Reflection.CLRExceptionHandler[],int,bool) 
0x0000025109de407b (Mono JIT Code) [CLRMethodBodyBase.cs:84] dotnow.Reflection.CLRMethodBodyBase:ExecuteMethodBody (dotnow.Runtime.ExecutionEngine,dotnow.Runtime.ExecutionFrame) 
0x000002510cf327cb (Mono JIT Code) [ExecutionMethod.cs:106] dotnow.Runtime.ExecutionMethod:DelegateInvoke (object) 
0x000002510cf32383 (Mono JIT Code) [CLRMethod.cs:295] dotnow.Reflection.CLRMethod/<>c__DisplayClass41_0:<CreateDelegate>b__0 () 
0x000002510cf31c00 (Mono JIT Code) [MonoBehaviourProxy.cs:265] UnityEngine.MonoBehaviourProxy:Start () 
0x00000250bfe39af8 (Mono JIT Code) (wrapper runtime-invoke) object:runtime_invoke_void__this__ (object,intptr,intptr,intptr)
0x00007ffe35c4f1e0 (mono-2.0-bdwgc) [mini-runtime.c:2849] mono_jit_runtime_invoke 
0x00007ffe35bd2ac2 (mono-2.0-bdwgc) [object.c:2921] do_runtime_invoke 
0x00007ffe35bdbb1f (mono-2.0-bdwgc) [object.c:2968] mono_runtime_invoke 
0x00007ff66330cd34 (Unity) scripting_method_invoke
0x00007ff663308011 (Unity) ScriptingInvocation::Invoke
0x00007ff6632dbfad (Unity) MonoBehaviour::InvokeMethodOrCoroutineChecked
0x00007ff6632dc0d8 (Unity) MonoBehaviour::InvokeMethodOrCoroutineChecked
0x00007ff6632d6ce2 (Unity) MonoBehaviour::DelayedStartCall
0x00007ff662e4bfa4 (Unity) DelayedCallManager::Update
0x00007ff663044099 (Unity) `InitPlayerLoopCallbacks'::`2'::EarlyUpdateScriptRunDelayedStartupFrameRegistrator::Forward
0x00007ff66302c4dc (Unity) ExecutePlayerLoop
0x00007ff66302c5b3 (Unity) ExecutePlayerLoop
0x00007ff663033059 (Unity) PlayerLoop
0x00007ff663cefa91 (Unity) PlayerLoopController::UpdateScene
0x00007ff663cdbcef (Unity) PlayerLoopController::EnterPlayMode
0x00007ff663ceb11d (Unity) PlayerLoopController::SetIsPlaying
0x00007ff663cedf87 (Unity) Application::TickTimer
0x00007ff6641334d1 (Unity) MainMessageLoop
0x00007ff664137521 (Unity) WinMain
0x00007ff6657c5752 (Unity) __scrt_common_main_seh
0x00007ffec87f7034 (KERNEL32) BaseThreadInitThunk
0x00007ffec9b82651 (ntdll) RtlUserThreadStart

At instruction: Ldc_I4_1
0x00007ff6633c8bdc (Unity) StackWalker::GetCurrentCallstack
0x00007ff6633cffd9 (Unity) StackWalker::ShowCallstack
0x00007ff66412cd9c (Unity) GetStacktrace
0x00007ff664b75e83 (Unity) DebugStringToFile
0x00007ff6634802f6 (Unity) DebugLogHandler_CUSTOM_Internal_Log
0x0000025034e9d30b (Mono JIT Code) (wrapper managed-to-native) UnityEngine.DebugLogHandler:Internal_Log (UnityEngine.LogType,UnityEngine.LogOption,string,UnityEngine.Object)
0x0000025034e9d1ab (Mono JIT Code) UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
0x0000025109dcacee (Mono JIT Code) UnityEngine.Logger:Log (UnityEngine.LogType,object)
0x000002510cf359ea (Mono JIT Code) UnityEngine.Debug:LogError (object)
0x0000025109de4e73 (Mono JIT Code) [ExecutionEngine.cs:183] dotnow.Runtime.ExecutionEngine:Execute (dotnow.AppDomain,dotnow.Runtime.ExecutionFrame,dotnow.Runtime.CIL.CILOperation[],dotnow.Reflection.CLRExceptionHandler[],int,bool) 
0x0000025109de407b (Mono JIT Code) [CLRMethodBodyBase.cs:84] dotnow.Reflection.CLRMethodBodyBase:ExecuteMethodBody (dotnow.Runtime.ExecutionEngine,dotnow.Runtime.ExecutionFrame) 
0x000002510cf327cb (Mono JIT Code) [ExecutionMethod.cs:106] dotnow.Runtime.ExecutionMethod:DelegateInvoke (object) 
0x000002510cf32383 (Mono JIT Code) [CLRMethod.cs:295] dotnow.Reflection.CLRMethod/<>c__DisplayClass41_0:<CreateDelegate>b__0 () 
0x000002510cf31c00 (Mono JIT Code) [MonoBehaviourProxy.cs:265] UnityEngine.MonoBehaviourProxy:Start () 
0x00000250bfe39af8 (Mono JIT Code) (wrapper runtime-invoke) object:runtime_invoke_void__this__ (object,intptr,intptr,intptr)
0x00007ffe35c4f1e0 (mono-2.0-bdwgc) [mini-runtime.c:2849] mono_jit_runtime_invoke 
0x00007ffe35bd2ac2 (mono-2.0-bdwgc) [object.c:2921] do_runtime_invoke 
0x00007ffe35bdbb1f (mono-2.0-bdwgc) [object.c:2968] mono_runtime_invoke 
0x00007ff66330cd34 (Unity) scripting_method_invoke
0x00007ff663308011 (Unity) ScriptingInvocation::Invoke
0x00007ff6632dbfad (Unity) MonoBehaviour::InvokeMethodOrCoroutineChecked
0x00007ff6632dc0d8 (Unity) MonoBehaviour::InvokeMethodOrCoroutineChecked
0x00007ff6632d6ce2 (Unity) MonoBehaviour::DelayedStartCall
0x00007ff662e4bfa4 (Unity) DelayedCallManager::Update
0x00007ff663044099 (Unity) `InitPlayerLoopCallbacks'::`2'::EarlyUpdateScriptRunDelayedStartupFrameRegistrator::Forward
0x00007ff66302c4dc (Unity) ExecutePlayerLoop
0x00007ff66302c5b3 (Unity) ExecutePlayerLoop
0x00007ff663033059 (Unity) PlayerLoop
0x00007ff663cefa91 (Unity) PlayerLoopController::UpdateScene
0x00007ff663cdbcef (Unity) PlayerLoopController::EnterPlayMode
0x00007ff663ceb11d (Unity) PlayerLoopController::SetIsPlaying
0x00007ff663cedf87 (Unity) Application::TickTimer
0x00007ff6641334d1 (Unity) MainMessageLoop
0x00007ff664137521 (Unity) WinMain
0x00007ff6657c5752 (Unity) __scrt_common_main_seh
0x00007ffec87f7034 (KERNEL32) BaseThreadInitThunk
0x00007ffec9b82651 (ntdll) RtlUserThreadStart

NotSupportedException: Specified method is not supported.
dotnow.Runtime.CIL.CILInterpreter.ExecuteInterpreted (dotnow.AppDomain domain, dotnow.Runtime.ExecutionEngine engine, dotnow.Runtime.ExecutionFrame& frame, dotnow.Runtime.CIL.CILOperation[]& instructions, dotnow.Reflection.CLRExceptionHandler[]& exceptionHandlers, dotnow.Runtime.ExecutionEngine+DebugFlags debugFlags) (at Assets/Plugins/dotnow/Scripts/Runtime/CIL/CILInterpreter.cs:2268)
dotnow.Runtime.ExecutionEngine.Execute (dotnow.AppDomain domain, dotnow.Runtime.ExecutionFrame frame, dotnow.Runtime.CIL.CILOperation[] methodInstructions, dotnow.Reflection.CLRExceptionHandler[] exceptionHandlers, System.Int32 initialInstructionIndex, System.Boolean debuggerResume) (at Assets/Plugins/dotnow/Scripts/Runtime/ExecutionEngine.cs:190)
dotnow.Reflection.CLRMethodBodyBase.ExecuteMethodBody (dotnow.Runtime.ExecutionEngine engine, dotnow.Runtime.ExecutionFrame frame) (at Assets/Plugins/dotnow/Scripts/Reflection/CLRMethodBodyBase.cs:84)
dotnow.Runtime.CIL.CILInterpreter.ExecuteInterpreted (dotnow.AppDomain domain, dotnow.Runtime.ExecutionEngine engine, dotnow.Runtime.ExecutionFrame& frame, dotnow.Runtime.CIL.CILOperation[]& instructions, dotnow.Reflection.CLRExceptionHandler[]& exceptionHandlers, dotnow.Runtime.ExecutionEngine+DebugFlags debugFlags) (at Assets/Plugins/dotnow/Scripts/Runtime/CIL/CILInterpreter.cs:2646)
dotnow.Runtime.ExecutionEngine.Execute (dotnow.AppDomain domain, dotnow.Runtime.ExecutionFrame frame, dotnow.Runtime.CIL.CILOperation[] methodInstructions, dotnow.Reflection.CLRExceptionHandler[] exceptionHandlers, System.Int32 initialInstructionIndex, System.Boolean debuggerResume) (at Assets/Plugins/dotnow/Scripts/Runtime/ExecutionEngine.cs:190)
dotnow.Reflection.CLRMethodBodyBase.ExecuteMethodBody (dotnow.Runtime.ExecutionEngine engine, dotnow.Runtime.ExecutionFrame frame) (at Assets/Plugins/dotnow/Scripts/Reflection/CLRMethodBodyBase.cs:84)
dotnow.Runtime.ExecutionMethod.DelegateInvoke (System.Object obj) (at Assets/Plugins/dotnow/Scripts/Runtime/ExecutionMethod.cs:106)
dotnow.Reflection.CLRMethod+<>c__DisplayClass41_0.<CreateDelegate>b__0 () (at Assets/Plugins/dotnow/Scripts/Reflection/CLRMethod.cs:295)
UnityEngine.MonoBehaviourProxy.Start () (at Assets/Plugins/dotnow/Scripts/Integration/Unity/MonoBehaviourProxy.cs:265)

```

**Comment by scottyboy805** [comment]
I has a quick look at this but can't see anything obvious. It looks like the stack gets corrupted at some stage but more investigation is required. I will check it again when I have some more free time.

**Comment by scottyboy805** [comment]
I finally got around to looking at this again and it seems like it no longer reproduces.
Possibly it was fixed in a recent commit like [7199cdb](https://github.com/scottyboy805/dotnow-interpreter/commit/7199cdbb9cf4ccd2c21569be2b29b2759f555a4f) or [1153fb5](https://github.com/scottyboy805/dotnow-interpreter/commit/1153fb52070c989158230cb502ede8baeafff90f)
